cmd/sourced/compose/workdir/common.go

```diff
@@ -191,26 +191,38 @@ func ListPaths() ([]string, error) {
 	return res, nil
 }
 
-// RemovePath removes working directory by removing required files
+// RemovePath removes working directory by removing its files,
 // and recursively removes directories up to the workdirs root as long as they are empty
 func RemovePath(path string) error {
 	workdirsRoot, err := workdirsPath()
 	if err != nil {
 		return err
 	}
 
-	for _, f := range RequiredFiles {
-		if err := os.Remove(filepath.Join(path, f)); err != nil {
-			return errors.Wrap(err, "could not remove from workdir directory")
-		}
-	}
-
 	for {
-		files, err := ioutil.ReadDir(path)
+		containsDirs := false
+		err := filepath.Walk(path, func(item string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				containsDirs = true
+				return nil
+			}
+
+			if err := os.Remove(item); err != nil {
```

Comment:
you delete files not only from the workdir but from other workdirs too (which happen to be inside this one)

Reply:
wow! that's true!
I assumed that our current behavior was right, so I just reproduced it, but it is not.
Will fix; thanks for your deep review !!!

Reply:
wow! that's true! many thanks for your deep review !!!

Reply:
0861efa61e9b1f918cbe2f57d344416afa56e616 Now `RemovePath` only removes its stuff